### PR TITLE
Add Scala.js support

### PR DESCRIPTION
**Pull request by exoego**
Addresses https://github.com/sbt/sbt/issues/4207

### CODE REVIEW

File: src/main/scala/sbt/ProjectMatrix.scala [hunk 1]
```diff
@@ -199,19 +205,26 @@ object ProjectMatrix {
     override def jvmPlatform(settings: Setting[_]*): ProjectMatrix =
       custom(jvmIdSuffix, jvmDirectorySuffix, Nil, { _.settings(settings) })
 
-    override def jvm: ProjectFinder = new ProjectFinder {
+    override def jsPlatform(settings: Setting[_]*): ProjectMatrix =
+      custom(jsIdSuffix, jsDirectorySuffix, Nil, { _.settings(settings) })
```
Comment (eed3si9n) [inline src/main/scala/sbt/ProjectMatrix.scala]: Where is it adding actual JS support?

Comment (exoego) [reply src/main/scala/sbt/ProjectMatrix.scala]: I might miss some context for this feature.
Do you think some treatment specific to Scala.js required?

Comment (eed3si9n) [reply src/main/scala/sbt/ProjectMatrix.scala]: I was expecting it to actually configure the generated subproject to build using Scala.js.

Comment (eed3si9n) [reply src/main/scala/sbt/ProjectMatrix.scala]: I have an idea for detecting the Scala.JS plugin without adding actual dependency.

**Pull request merged**